pkg/skaffold/deploy/kubectl/kubectl.go

```diff
@@ -110,21 +122,27 @@ func (k *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Art
 	if len(manifests) == 0 {
 		return nil, nil
 	}
-
+	ctx, endTrace := instrumentation.StartTrace(ctx, "Deploy_CollectNamespaces")
 	namespaces, err := manifests.CollectNamespaces()
 	if err != nil {
 		event.DeployInfoEvent(fmt.Errorf("could not fetch deployed resource namespace. "+
 			"This might cause port-forward and deploy health-check to fail: %w", err))
 	}
+	endTrace()
 
+	ctx, endTrace = instrumentation.StartTrace(ctx, "Deploy_WaitForDeletions")
```

Comment:
Do we need to worry that `ctx` (the argument) is actually the ctx from the previous `StartTrace()`, and not the ctx passed into this function?  Does the `endTrace()` re-link the context?

Reply:
I have changed the `Deploy_` and `doDev_` calls which had multiple `StartTrace` calls in a single function and passed `ctx` down the chain to properly create a new ctx (`childCtx`) and pass that to the relevant children for each command s.t. each direct trace in the function uses the initial root ctx